lightning/src/ln/channelmanager.rs

```diff
@@ -6887,6 +6959,8 @@ impl<'a, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref>
 								hash_map::Entry::Vacant(entry) => {
 									let path_fee = path.get_path_fees();
 									entry.insert(PendingOutboundPayment::Retryable {
+										retry_strategy: Retry::Attempts(0),
```

Comment:
Should we have some reasonable default? `Retry::Attempts(0)` would mean no retries. In what situations would we hit this case?

Reply:
We hit this while deserializing `ChannelManager` if a monitor has some htlcs that the `ChannelManager` forgot about due to being behind the monitor on shutdown, IIRC. Idea is that we never want to retry payments after restart, because the payments are probably stale anyway

Reply:
I believe we chose not to retry after restart in `InvoicePayer` because we didn't want to have something else for the user to serialized. But now that the retry logic is in `ChannelManager`, that rational is no longer applicable. We can still use the no-retry behavior for older serializations, though.

Reply:
> Maybe it's a consideration for follow-up, but it does seem like payments would be stale after restart? And even if they weren't, it seems like the sanest default would be to not retry 🤔

Restart might just be failing over to a new server, FWIW. Or maybe for mobile switching back and forth between apps?

I think we should at least decided what we'd like to happen to avoid unnecessary churn in the serialization format.

> I think the invoice payer differed a bit since it has a static retry strategy config, which we don't have here

It was mostly because we didn't want to serialize it. If we serialized, the static strategy would have been serialized along with it, I'd imagine.